components/org.wso2.carbon.crypto.provider.hsm/src/main/java/org/wso2/carbon/crypto/provider/hsm/cryptoprovider/operators/Cipher.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.crypto.provider.hsm.cryptoprovider.operators;
+
+import iaik.pkcs.pkcs11.Mechanism;
+import iaik.pkcs.pkcs11.Session;
+import iaik.pkcs.pkcs11.TokenException;
+import iaik.pkcs.pkcs11.objects.Key;
+import iaik.pkcs.pkcs11.wrapper.PKCS11Constants;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.crypto.api.CryptoException;
+import org.wso2.carbon.crypto.provider.hsm.cryptoprovider.exception.HSMCryptoException;
+
+/**
+ * This class is responsible for carrying out encrypt/decrypt operations.
+ */
+public class Cipher {
```

Comment:
Is this name meaningful ?. A class which is responsible for encryption and decryption is named as Cipher ?

Reply:
Maybe CipherUtil ?